src/layers/ethernet/mod.rs

```diff
@@ -36,12 +39,29 @@ pub(crate) fn register_defaults() -> Result<(), Error> {
 /// A Layer that would handle subsequent decoding for a given Ethertype, should register itself
 /// by calling this function. For example [`crate::layers::ipv4`] would call `register_ethertype`
 /// with [`EtherType`] value of 0x0800, passing the creator function for that layer.
-pub fn register_ethertype(eth_type: EtherType, layer: LayerCreatorFn) -> Result<(), Error> {
-    let mut map = get_ethertypes_map().write().unwrap();
-    if map.contains_key(&eth_type) {
-        return Err(Error::RegisterError(format!("ether_type: {}", eth_type)));
+pub fn register_ethertype(
+    eth_type: EtherType,
+    name: &str,
+    layer: LayerCreatorFn,
+) -> Result<(), Error> {
+    {
+        let mut map = get_ethertypes_map().write().unwrap();
+        if map.contains_key(&eth_type) {
+            return Err(Error::RegisterError(format!("ether_type: {}", eth_type)));
+        }
+        map.insert(eth_type, layer);
+    }
+
+    {
+        let mut inv_map = get_inv_ethertypes_map().write().unwrap();
```

Comment:
Also, wondering if it's a good idea to keep all lower case strings to map like `arp`, `ipv6` (user can pass whatever, we `lowercase` it to store and use to `get` as well. This way the user does not have to "remember" whether it's upper case or lowercase or any mixed case.

Reply:
I'd used `Layer::name()` for this, should I be using `Layer::short_name()` instead?

Reply:
> I'd used `Layer::name()` for this, should I be using `Layer::short_name()` instead?

Yeah that would work too!!

Reply:
In that case, I think it makes sense to write some sort of wrapper around `std::collections::Hashmap` to allow case-insensitive mapping(or use an existing library like [case_insensitive_hashmap](https://docs.rs/case_insensitive_hashmap/latest/case_insensitive_hashmap/))

Reply:
Hey, I just realized that you wanted to do this for a more user-friendly UI. But this is meant to be an internal API and the Key should ideally be the `.name()` attribute of a layer. The user interface will be built later, to parse a statement like "Ethernet() / Ipv6()", and that can have a case-insensitive map.